### ISSUE TITLE
Incorrect unit test ComparisonFailure messages

### ISSUE THREAD

**Issue by cr3ativ3** [DroidsOnRoids/jspoon#56]
Most of jspoon's unit tests (the ones not written by me) has assertEquals(arg1, arg2) with arguments switched places. They have first _actual_ then _expected_, when it should be first _expected_ then _actual_. 
That results in **misleading** junit's error messages upon test failure:
```
String actual = "actual";
assertEquals(actual, "expected");

org.junit.ComparisonFailure: expected:<[actual]> but was:<[expected]>
```

**Comment by koral--** [comment]
Good catch. We migth consider switching e.g. to assertJ. It uses `assertThat("actual").isEqualTo("expected")` so it is harder to make such mistakes.

**Comment by cr3ativ3** [comment]
I don't know if its any harder to write `assertThat("expected").isEqualTo("actual")`. Though assertj's builder style assertions does make it easier if do multiple chained evaluations on the same instance, but haven't seen that many use-cases in jspoon.
